### PR TITLE
[Icons] Add metafields type icons

### PR DESCRIPTION
**Pull request by leileu**
### WHY are these changes introduced?

We are currently using these icons in the Metafields Settings, admin, and OSE to for the following types:
- Color
- JSON
- Measurement (Dimension, Volume, Weight)
- Money

And in the code they are currently these are all custom.

This issue will relate to 
[#229](https://github.com/Shopify/metafield-issues/issues/229)
[#7152](https://github.com/Shopify/polaris/issues/7152) 

### WHAT is this pull request doing?

Adding the following Metafield type icons (in Minor only):
- Color
- JSON
- Measurement
- Money

Screen Shot 2022-09-12 at 3 25 23 PM

### CODE REVIEW

**Comment by selenehinkley** [comment]
Hello 😃 @kyledurand asked me to take a little look at the naming of the icons so I took a glance at our [icon library](https://polaris.shopify.com/icons) to compare. 
- ✅ money seems fine because `cash` is already taken for dollar/euro/pound/rupee denominations. I thought of other options, but `bill` might be confused with billing. It would've been nice to have something more specific since "money" could apply to more than one method of payment and we have an icon for `card`, but I had trouble coming up with something other than `cash`.
- JSON seemed a bit specific for curly brackets (excuse me if I'm lacking subject matter expertise here). We use the term `code` for anything with curly brackets already. Would the existing code icon not work? Or is there a `code (x)` term we could use to be consistent? Example: `code JSON`
Screen Shot 2022-09-14 at 1 57 53 PM
Screen Shot 2022-09-14 at 1 58 06 PM

- ✅ `color dropper` makes sense and there aren't any conflicts 

- ✅ `measurement` makes sense and there aren't any conflicts

**Comment by leileu** [comment]
Hi @selenehinkley You make a good point on JSON. These icons were created before my time, but I did have the same question about these if they are duplicates of existing ones. Is Code icon used anywhere else? Money is also something I can relook at.

@kyledurand Let me pause on this PR and review it again!

**Comment by leileu** [comment]
> 

@selenehinkley Also wanted to note, the suggestions (Code, Cash) are all Majors where I'm trying to add Minors for Metafields.

**Review by sarahill — APPROVED** [review]
✅

**Review by kyledurand — APPROVED** [review]
LGTM. Thanks @leileu

**Comment by leileu** [comment]
@kyledurand @sarahill I revised some of the icons based on our slack thread and pushed the changes back to the branch. Could someone review? thanks!

File: .changeset/fair-dryers-wave.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris-icons': major
```
Comment (kyledurand) [inline .changeset/fair-dryers-wave.md]: This shouldn't a major version change. I think we can merge these two change sets into one.

Comment (leileu) [reply .changeset/fair-dryers-wave.md]: Oh right, I think it's for both. Do i need to correct this or did you already?

Comment (kyledurand) [reply .changeset/fair-dryers-wave.md]: All fixed up! I think you can 🚢 this PR now 👍

Comment (leileu) [reply .changeset/fair-dryers-wave.md]: Thanks Kyle!

Comment (leileu) [reply .changeset/fair-dryers-wave.md]: @kyledurand how long does it take to show up on polaris page? I merged it but dont' see it?

Comment (kyledurand) [reply .changeset/fair-dryers-wave.md]: They'll appear after the next Polaris release. You can read more about the release process [here](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md). I'm not aware of any upcoming releases but we've been averaging one per week recently.

**Pull request merged**